validator/src/bin/solana-test-validator.rs

```diff
@@ -304,7 +304,14 @@ fn main() {
         });
     }
 
-    let mut ledger_fd_lock = fd_lock::RwLock::new(fs::File::open(&ledger_path).unwrap());
+    let lockfile = ledger_path.join("ledger.lock");
+    let mut ledger_fd_lock = fd_lock::RwLock::new(
+        fs::OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(&lockfile)
+            .unwrap(),
+    );
```

Comment:
This duplication between `solana-test-validator` and `solana-validator` is now a little too big for my preference.  How about a shared function in validator/src/lib.rs for this